### DISCUSSION TITLE
java sdk，When querying and adding, the precision can only be float？

### DISCUSSION THREAD

**Discussion by qinjie1314520** [milvus-io/milvus#17325] [Q&A and General discussion] [answered]
I am using the Java SDK to add new vectors, but in Java, float precision is limited,

This leads to insufficient vector precision after new entry

**Comment by yhmo** [comment, marked as answer]
ANN search is approximate nearest neighbor searching. It is approximate not accuracy, most of ANN libraries use float as input instead of double. Milvus store vectors as float values, compute vectors as float values, so the Java sdk only allow input vector as `List<Float>`.
By the way, the python sdk accept vectors as double precision, but in fact these values are converted to single precision(float) in server side.

**Reply by qinjie1314520** [reply]
I see. Thank you for your guidance
